### PR TITLE
[MBL-16139][Student] Share Extension progress

### PR DESCRIPTION
**Pull request by hermannakos**
refs: MBL-16139
affects: Student
release note: Updated the Share Extension workflow.

test plan: Smoke test the whole Share Extension workflow.

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [ ] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressDialogViewModel.kt [hunk 1]
```diff
@@ -1,7 +1,180 @@
...
+    @DrawableRes private fun getIconDrawableRes(contentType: String): Int {
+        return when {
+            contentType.contains("image") -> R.drawable.ic_image
+            contentType.contains("video") -> R.drawable.ic_media
+            contentType.contains("pdf") -> R.drawable.ic_pdf
+            else -> R.drawable.ic_attachment
+        }
+    }
+
+    private fun humanReadableByteCount(bytes: Long): String {
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/shareextension/progress/ShareExtensionProgressDialogViewModel.kt:165]: I have seen this already somewhere. Maybe we could make this an extension function.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Pull request merged**